src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,145 @@
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import ChatSchema from '../../../database/schema/ChatSchema';
+import UserSchema from '../../../database/schema/UserSchema';
+import {DEFAULT_PROFILE_PIC_PATH} from '../../../utils/constants';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import useChatUtilsHook from './useChatUtilsHook';
+import {initChatFromPost, initChatFromPostAnon} from '../../../service/chat';
+import {getChannelListInfo} from '../../../utils/string/StringUtils';
+
+type ChannelCategory = 'SIGNED' | 'ANONYMOUS';
+
+const useDMMessage = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {goToChatScreen} = useChatUtilsHook();
+
+  const saveChannelList = async (
+    channel,
+    type: ChannelCategory,
+    builtChannelData: {
+      better_channel_member: any;
+      members: any;
+    }
+  ) => {
+    try {
+      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
+      let userTarget = channel?.better_channel_members.find(
+        (data) => data?.user?.id !== signedProfileId
+      );
+
+      if (type === 'ANONYMOUS') {
+        userTarget = channel?.better_channel_members.find(
+          (data) => data?.user?.id !== anonProfileId
+        );
+      }
+
+      const {
+        anonUserInfoEmojiName,
+        anonUserInfoColorCode,
+        anonUserInfoColorName,
+        anonUserInfoEmojiCode,
+        channelName,
+        channelImage
+      } = getChannelListInfo(builtChannelData, signedProfileId, anonProfileId);
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: channelName,
+        image: channelImage
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId; // change to use getChannelListInfo
+      channel.targetName = chatName?.name; // change to use getChannelListInfo
+      channel.targetImage = chatName?.image; // change to use getChannelListInfo
+
+      channel.channel = {...channel};
+      console.log('data', {
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      });
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members, {
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      });
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `saveChannelList` seems to be doing a lot of things: finding user targets, getting channel list info, setting up chat types and names, updating the channel object, logging data, creating a new channel list from API data, saving the channel list if it's the latest, refreshing certain parts of the app, and navigating to the chat screen. This is a violation of the Single Responsibility Principle (SRP). Consider breaking down this function into smaller, more manageable functions each with a single responsibility. This will improve readability, maintainability, and testability of your code.

```diff
- const saveChannelList = async (
-   channel,
-   type: ChannelCategory,
-   builtChannelData: {
-     better_channel_member: any;
-     members: any;
-   }
- ) => {
-   // ... existing code ...
- };
+ const findUserTarget = (channel, type) => {
+   // logic for finding user target
+ };
+
+ const getChatTypeAndName = (type, channelName, channelImage) => {
+   // logic for getting chat type and name
+ };
+
+ const updateChannelObject = (channel, signedProfileId, chatName) => {
+   // logic for updating channel object
+ };
+
+ const createAndSaveChannelList = async (channel, chatType, localDb, refresh) => {
+   // logic for creating and saving channel list
+ };
+
+ const saveChannelList = async (
+   channel,
+   type: ChannelCategory,
+   builtChannelData: {
+     better_channel_member: any;
+     members: any;
+   }
+ ) => {
+   const userTarget = findUserTarget(channel, type);
+   const {chatType, chatName} = getChatTypeAndName(type, channelName, channelImage);
+   updateChannelObject(channel, signedProfileId, chatName);
+   await createAndSaveChannelList(channel, chatType, localDb, refresh);
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->